src/cdk/env.ts

```diff
@@ -0,0 +1,63 @@
+import { MiraConfig, Account } from '../config/mira-config'
+
+/** @ignore - Excluded from documentation generation. */
+// eslint-disable-next-line
+const minimist = require('minimist')
+
+/** @ignore - Excluded from documentation generation. */
+const args = minimist(process.argv.slice(2))
+
+/**
+ * @class Responsible for establishing the Mira environment.  This includes
+ * assisting with the bootstrapping process (reading and parsing CLI args) and
+ * the local system configuration including AWS credentials.
+ */
+export default class MiraEnv {
+    env: Account
+    static instance: MiraEnv
+    constructor () {
+      if (!MiraEnv.instance) {
+        MiraEnv.instance = this
+      } else if (args.env !== 'test') {
+        console.warn('MiraEnv was instantiated twice outside a testing environment' +
+              '.  This will likely cause unknown behavior.')
+      }
+    }
+
+    initialize (): void {
+      this.parseFile()
+      this.parseEnv()
+    }
+
+    /**
+ * Parses the environment variable.
+ */
+    parseEnv (): Account {
+      if (!args.env && process.env.NODE_ENV) {
+        args.env = process.env.NODE_ENV
+      } else if (!args.env) {
+        console.warn('Warning: Environment not specified, defauling to dev.')
+        args.env = 'dev'
+      }
+      // TODO: Perhaps we migrate this fn to this class?
+      this.env = MiraConfig.getEnvironment(args.env)
+      return this.env
+    }
+
+    /**
+     * Parses the file variable.
+     */
+    parseFile (): string {
+      if (!args.file) {
+        console.warn('You must specify a --file argument when using the ' +
+                'deploy or undeploy command within mira.')
+        // TODO: Specify exit code.
+        process.exit(1)
+      }
+      if (Array.isArray(args.file)) {
+        return args.file
+      }
+      return args.file.split(',')
+    }
+}
+new MiraEnv()
```

Comment:
Wouldn't be better to export the instance as we want the `MiraEnv` to be a singleton, so we don't have to do a check and the static instance property on `MiraEnv` class (similarly to `MiraConfig`)?

Reply:
I  roughly agree.  I think this is me trying to implement prior feedback not to use singletons for testing purposes.  I think so long as we export the classname as well, seems like it would be fine to make the instance the default export.